test/validation/validators/neighbor_validator_test.py

```diff
@@ -0,0 +1,111 @@
+from otri.validation.validators.neighbor_validator import NeighborValidator
+from otri.validation.exceptions import NeighborWarning
+from otri.filtering.stream import Stream
+from .. import find_error
+
+import unittest
+from typing import Callable, Iterable, Mapping
+
+
+class NeighborValidatorTest(unittest.TestCase):
+
+    def template(self, find: Callable, test_data: Iterable, expected: Iterable, t_range: int, limits: Mapping):
+        '''
+        Parameters:
+
+            find : Callable
+                Function converting an output list into some evaluable result.
+
+            test_data : Iterable
+                The test data to put in the input Streams. Must be a list of input datasets.
+
+            expected : Iterable
+                The expected results after passing through `find`. Must be a list of expected outputs.
+                Must be the same size as test_data.
+
+            t_range : int
+                The time range for the validation.
+
+            limits : Mapping
+                Mapping of the keys to check and their discrepancy limits.
+        '''
+        inputs = [Stream(batch, is_closed=True) for batch in test_data]
+        outputs = [Stream() for _ in test_data]
+
+        f = NeighborValidator([str(i) for i in range(len(inputs))],
+                              [str(-i) for i in range(len(outputs))],
+                              t_range,
+                              limits)
+        f.setup(inputs, outputs, dict())
+
+        while not f._are_outputs_closed():
+            f.execute()
+
+        results = [list(output) for output in f._get_outputs()]
+        prepared_outputs = [find(output) for output in results]
+
+        for i in range(len(prepared_outputs)):
```

Comment:
Codacy found an issue: [Consider using enumerate instead of iterating with range and len](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=6277039)